examples/eval_p3/eval.py

```diff
@@ -0,0 +1,109 @@
+import logging
+from dataclasses import dataclass
+from typing import Union
+
+from rouge_score.scoring import Score
+
+from tango import Format, JsonFormat, Step
+from tango.common import DatasetDict
+from tango.common.logging import make_tqdm
+
+logger = logging.getLogger(__name__)
+tqdm = make_tqdm(logger)
+
+
+@dataclass
+class RougeScore:
```

Comment:
This class adds a lot of code to the example that seems overkill for just aggregating a metric. Seems like the `Score` named tuple from `rouge_score.scoring` is good enough already. For example, you could aggregate a bunch of `Score` objects like this:

```python
scores = [Score(0.5, 0.4, 0.6), Score(0.1, 0.3, 0.2)]
Score(*[sum(x) / len(scores) for x in zip(*scores)])
```

Reply:
I keep three of these score tuples around, and I do four arithmetic operations with them, so I'd end up with 12 lines like `Score(*[sum(x) / len(scores) for x in zip(*scores)])`. Though I can probably condense two of the operations, so then it's 9.

I'd argue that while mine has more lines, it's easier to read. I could remove some of the actions (`__imul__` is never used, for example), but this is not a test, it's an example of how to write code.

Reply:
None of this code is still around after switching to torchmetrics.